### PR TITLE
[WIP] Adds toggling filter button for iOS Collections

### PR DESCRIPTION
**Pull request by ashleyjelks**
https://artsyproduct.atlassian.net/browse/FX-1765

Adds an unstyled filter button for opening filter artworks modal that only renders when collection artworks grid is within the viewport.

Kapture 2020-01-30 at 18 23 04

### CODE REVIEW

File: src/lib/Scenes/Collection/Collection.tsx [hunk 1]
```diff
@@ -66,13 +72,25 @@ export class Collection extends Component<CollectionProps, CollectionState> {
...
-    const { sections } = this.state
+    const { isArtworkGridVisible, sections } = this.state
 
     return (
       <Theme>
-        <Box>
+        <View style={{ flex: 1 }}>
           <FlatList
+            onViewableItemsChanged={this.onViewableItemsChanged}
+            viewabilityConfig={this.viewabilityConfig}
```
Comment (ds300) [inline src/lib/Scenes/Collection/Collection.tsx]: :+1: good find!

File: src/lib/Scenes/Collection/Collection.tsx [hunk 2]
```diff
@@ -25,8 +28,11 @@ interface CollectionState {
 export class Collection extends Component<CollectionProps, CollectionState> {
   state = {
     sections: [],
+    isArtworkGridVisible: false,
+  }
+  viewabilityConfig = {
+    viewAreaCoveragePercentThreshold: 5,
```
Comment (anandaroop) [inline src/lib/Scenes/Collection/Collection.tsx]: Sooo in plain English this means the artwork grid must cover 5% of the viewport before it is considered visible? (I _think_ so, but it was a little hard to tell from the gif bc of that yellow alert.)

Comment (ashfurrow) [reply src/lib/Scenes/Collection/Collection.tsx]: I _think_ so. [I found the docs](https://facebook.github.io/react-native/docs/flatlist#viewareacoveragepercentthreshold) and it appears to be the case. Seems like we want the filter button to appear only when this is the case, which makes sense 👍 In any case, a short comment would be super-helpful here!

**Comment by ashleyjelks** [comment]
@anandaroop @ds300 @ashfurrow I [combined this PR](https://github.com/artsy/emission/pull/2073) with #2067 so I'm going to close this for now. But will add a comment about the visibility percentage you mentioned here!
